tests/utils/profilehandler/profile_handler.go

```diff
@@ -681,3 +681,157 @@ func CreateClusterByProfile(profile *Profile, client *rosacli.Client, waitForClu
 	}
 	return description, err
 }
+
+func WaitForClusterUninstalled(client *rosacli.Client, cluster string, timeoutMin int) error {
+
+	endTime := time.Now().Add(time.Duration(timeoutMin) * time.Minute)
+	for time.Now().Before(endTime) {
+		output, err := client.Cluster.DescribeCluster(cluster)
+		desc, err := client.Cluster.ReflectClusterDescription(output)
+
+		if err != nil {
+			return err
+		}
+		if strings.Contains(output.String(), fmt.Sprintf("There is no cluster with identifier or name '%s'", cluster)) {
+			log.Logger.Infof("Cluster %s has been deleted.", cluster)
+			return nil
+		}
+		if strings.Contains(desc.State, con.Uninstalling) {
+			time.Sleep(2 * time.Minute)
+			continue
+		}
+		return fmt.Errorf("Cluster %s is in status of %s which won't be deleted, stop waiting", cluster, desc.State)
+	}
+	return fmt.Errorf("timeout for waiting for cluster deletion finished after %d mins", timeoutMin)
+}
+
+func DestroyClusterByProfile(profile *Profile, client *rosacli.Client, waitForClusterUninstall bool) (errors []error) {
+	var (
+		ud                 *UserData
+		cd                 *ClusterDetail
+		clusterService     rosacli.ClusterService
+		errDeleteCluster   error
+		rosaClient         *rosacli.Client
+		ocmResourceService rosacli.OCMResourceService
+	)
+	region := profile.Region
+	rosaClient = rosacli.NewClient()
+	ocmResourceService = rosaClient.OCMResource
+
+	// get cluster info from cluster detail file
+	cd, err := ParserClusterDetail()
+	if err != nil {
+		errors = append(errors, err)
+		return
+	}
+	// get user data from resource file
+	ud, err = ParseUserData()
+	if err != nil {
+		errors = append(errors, err)
+		return
+	}
+
+	// delete cluster
+	clusterService = client.Cluster
+	output, errDeleteCluster := clusterService.DeleteCluster(cd.ClusterID, "-y")
+	if errDeleteCluster != nil {
+		if strings.Contains(output.String(), fmt.Sprintf("There is no cluster with identifier or name '%s'", cd.ClusterID)) {
+			log.Logger.Infof("Cluster %s is not existed.", cd.ClusterID)
+		} else {
+			log.Logger.Errorf("Error happened when delete cluster: %s", output.String())
+			errors = append(errors, errDeleteCluster)
+			return
+		}
+	} else {
+		if waitForClusterUninstall {
+			log.Logger.Infof("Waiting for the cluster %s to uninstalled", cd.ClusterID)
+			err = WaitForClusterUninstalled(client, cd.ClusterID, config.Test.GlobalENV.ClusterWaitingTime)
+			if err != nil {
+				log.Logger.Errorf("Error happened when waitling cluster uninstall: %s", err.Error())
+				errors = append(errors, err)
```

Comment:
If uninstallation is not finished, let's just return here. Other resources cannot be deleted.